test/e2e/performanceprofile/functests/12_hypershift/hypershift.go

```diff
@@ -0,0 +1,200 @@
+			Expect(err).ToNot(HaveOccurred())
+
+			NodePoolLabel := map[string]string{
+				fmt.Sprintf("%s/%s", "hypershift.openshift.io", "nodePool"): nodePools[1].Name,
+			}
+			workerRTNodes, err = nodes.GetByLabels(NodePoolLabel)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verifying the kubelet parameters were correctly updated for the worker nodes")
+			for _, node := range workerRTNodes {
+				By(fmt.Sprintf("Checking kubelet configuration for node %s", node.Name))
+
+				result, err := chkKubeletConfigFn(context.TODO(), &node)
+				Expect(err).ToNot(HaveOccurred(), "Failed to fetch kubelet configuration")
+
+				obj, err := manifestsutil.DeserializeObjectFromData([]byte(result), kubeletconfigv1beta1.AddToScheme)
+				Expect(err).ToNot(HaveOccurred(), "Failed to deserialize kubelet configuration")
+
+				kc, ok := obj.(*kubeletconfigv1beta1.KubeletConfiguration)
+				Expect(ok).To(BeTrue(), "Deserialized object is not of type KubeletConfiguration")
+
+				Expect(kc.ReservedSystemCPUs).To(Equal("0,3"), "ReservedSystemCPUs is not correctly set")
+				Expect(kc.TopologyManagerPolicy).To(Equal("best-effort"), "TopologyManagerPolicy is not correctly set")
+			}
+		})
+
+		AfterAll(func() {
+			By("Deleting the second Profile")
+			Expect(nodepools.DeattachTuningObject(context.TODO(), testclient.ControlPlaneClient, secondProfile, nodePools[1])).To(Succeed())
+
+			By("Waiting for the second node pool configuration to start updating after profile deletion")
+			err = nodepools.WaitForUpdatingConfig(context.TODO(), testclient.ControlPlaneClient, np.Name, np.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for the second node pool configuration to be ready after profile deletion")
+			err = nodepools.WaitForConfigToBeReady(context.TODO(), testclient.ControlPlaneClient, np.Name, np.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			PrintNodePoolProfiles(nodePools)
+		})
+	})
+})
+
+func ListNodePools(ctx context.Context, c client.Client, hostedClusterName string) ([]*hypershiftv1beta1.NodePool, error) {
+	npList := &hypershiftv1beta1.NodePoolList{}
+	if err := c.List(ctx, npList); err != nil {
+		return nil, err
+	}
+
+	var nodePools []*hypershiftv1beta1.NodePool
+	for i := range npList.Items {
+		np := &npList.Items[i]
+		if np.Spec.ClusterName == hostedClusterName {
+			nodePools = append(nodePools, np)
+		}
+	}
+
+	if len(nodePools) == 0 {
+		return nil, fmt.Errorf("no nodePools found for hosted cluster %q", hostedClusterName)
+	}
+
+	return nodePools, nil
+}
+
+func PrintNodePoolProfiles(nodePools []*hypershiftv1beta1.NodePool) {
+	for _, np := range nodePools {
+		for _, tuningConfig := range np.Spec.TuningConfig {
+			testlog.Infof("NodePool %q is using profile: %q", np.Name, tuningConfig.Name)
+		}
+		if len(np.Spec.TuningConfig) == 0 {
+			testlog.Infof("NodePool %q does not have a tuningConfig profile", np.Name)
```

Comment:
when it does I think it'll be handy to know it though

Reply:
Info of profile for a nodepool is already getting logged. [Line](https://github.com/openshift/cluster-node-tuning-operator/pull/1184/files#diff-fce62ee3197b8d8d7e8e511b566166d711ca57a406e25bf356adf90f5a3abe74R199)